include/ISceneNode.h

```diff
@@ -621,6 +617,8 @@ namespace scene
 		//! Pointer to the parent
 		ISceneNode* Parent;
 
+		std::optional<ISceneNodeList::iterator> Iterator;
```

Comment:
How about `ChildIterator`? Logically it would also make more sense to have to below `Children`.
Also a quick documentation like with all other members would be nice.

Reply:
I've opted for `ThisIterator`, given that this isn't the iterator of some child, but of this node in the parent's child list.